cmd/datamon/cmd/cli_workshop_test.go

```diff
@@ -0,0 +1,494 @@
+// +build fuse_cli
+
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	gcsStorage "cloud.google.com/go/storage"
+	"github.com/oneconcern/datamon/internal"
+	"github.com/oneconcern/datamon/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type testFile struct {
+	data                        []byte
+	size                        int
+	listed, downloaded, mounted bool
+}
+
+// TestWorkshop goes through all the steps in our workshop.
+//
+// Test is run with spawned process on freshly built datamon binary, not test goroutine: no mocks, no tricks
+func TestWorkshop(t *testing.T) {
```

Comment:
Test Setup and Clean up can be simplified and include config and context creation.
I understand this is a script vs individual tests.
Might make sense to move the workshop to docs at some point.

Reply:
Created issues for that in github